src/commands/registerCommands.ts

```diff
@@ -139,49 +139,125 @@ for (const label of bindingTypes.keys()) {
-      const identifier = TextDocumentIdentifier.create(documentURI.toString());
-      const grammarURI = fileUri[0];
-      try {
-        const result = await commands.executeCommand(ServerCommandConstants.ASSOCIATE_GRAMMAR_INSERT, identifier, grammarURI.toString(), bindingType);
-        // Insert the proper syntax for binding
-        const lspTextDocumentEdit = <TextDocumentEdit>result;
-        const workEdits = new WorkspaceEdit();
-        for (const edit of lspTextDocumentEdit.edits) {
-          workEdits.replace(documentURI, languageClient.protocol2CodeConverter.asRange(edit.range), edit.newText);
-        }
-        workspace.applyEdit(workEdits); // apply the edits
-      } catch (error) {
-        window.showErrorMessage('Error during grammar binding: ' + error.message);
-      };
+/**
+ * Register commands used for associating grammar file (XSD,DTD) to a given XML file for command menu and CodeLens
+ *
+ * @param context the extension context
+ * @param languageClient the language server client
+ */
+ function registerAssociationCommands(context: ExtensionContext, languageClient: LanguageClient) {
+  // For CodeLens
+  context.subscriptions.push(commands.registerCommand(ClientCommandConstants.OPEN_BINDING_WIZARD, async (uriString: string) => {
+    grammarAssociationCommand(uriString, languageClient)
+  }));
+  // For command menu
+  context.subscriptions.push(commands.registerCommand(ClientCommandConstants.COMMAND_PALETTE_BINDING_WIZARD, async () => {
+    const uriString = window.activeTextEditor.document.fileName;
+    // Run check to ensure available grammar binding command should be executed, or if error is thrown
+    const canBind = await checkCanBindGrammar(window.activeTextEditor.document);
+    if (canBind) {
+      grammarAssociationCommand(uriString, languageClient)
     }
   }));
 
-}
\ No newline at end of file
+  // Setup listener for schema binding
+  context.subscriptions.push(activateCanBindGrammar({ xml: true, xsl: true }));
+ }
+
+  /**
+   * Perform a check to see if the current (XML) document has a grammar/schema bound to it on text editor open
+   *
+   * @param supportedLanguages the languages the listener should be active for
+   */
+  function activateCanBindGrammar(supportedLanguages: { [id: string]: boolean }): Disposable {
+
+    let disposables: Disposable[] = [];
+
+    // Call listener on text editor open to check if opened (XML) document had a bound grammar/schema
+    updateEnabledState();
+    window.onDidChangeActiveTextEditor(updateEnabledState, null, disposables);
+
+    async function updateEnabledState() {
+      // Check if the editor is open
+      let editor = window.activeTextEditor;
+      if (!editor) {
+        return;
+      }
+      // Check if the document language is supported
+      let document = editor.document;
+      if (!supportedLanguages[document.languageId]) {
+        return;
+      }
+
+      await checkCanBindGrammar(document);
```

Comment:
`await checkCanBindGrammar(document.uri);`

After that you canremove TextDocument import.